src/NuGet.Core/NuGet.Common/Logging/ExceptionLogger.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace NuGet.Common
+{
+    public class ExceptionLogger
+    {
+        private readonly IEnvironmentVariableReader _reader;
+
+        public ExceptionLogger(IEnvironmentVariableReader reader)
+        {
+            _reader = reader;
+        }
+
+        /// <summary>
+        /// Determines whether the full exception (including stack trace) should be displayed to
+        /// the user. In prerelease or dogfooding scenarios, it is useful to have a non-verbose
+        /// logging level but, in the case of an unhandled exception, print the full exception for
+        /// bug reporting.
+        /// </summary>
+        /// <returns>
+        /// True if the exception stack should be displayed to the user. False, otherwise.
+        /// </returns>
+        public bool ShouldShowStack()
+        {
+            var rawShowStack = _reader.GetEnvironmentVariable("NUGET_SHOW_STACK");
+            bool showStack;
+
+            if (!bool.TryParse(rawShowStack, out showStack))
+            {
+                showStack = false;
+            }
+
+            return showStack;
+        }
+
+        private static Lazy<ExceptionLogger> _instance = new Lazy<ExceptionLogger>(
```

Comment:
Shouldn't require this to be lazy.
`public static readonly ExceptionLogger Instance { get; } = new ExceptionLogger(new EnvironmentVariableWrapper());` should suffice